server/controllers/user.controller/authManagement.ts

```diff
@@ -0,0 +1,259 @@
+    res.json({
+      success: true,
+      message:
+        'If the email is registered with the editor, an email has been sent.'
+    });
+  } catch (err) {
+    if (process.env.NODE_ENV !== 'test') {
+      // don't log in test env
+      console.log(err);
+    }
+    res.json({ success: false });
+  }
+};
+
+/**
+ * - Method: `GET`
+ * - Endpoint: `/reset-password/:token`
+ * - Authenticated: `false`
+ * - Id: `UserController.validateResetPasswordToken`
+ *
+ * Description:
+ *   - The link in the Reset Password email, which contains a reset token that is valid for 1h
+ *   - If valid, the user will see a form to reset their password
+ *   - Else they will see a message that their token has expired
+ */
+export const validateResetPasswordToken: RequestHandler<
+  ResetOrUpdatePasswordRequestParams,
+  GenericResponseBody
+> = async (req, res) => {
+  const user = await User.findOne({
+    resetPasswordToken: req.params.token,
+    resetPasswordExpires: { $gt: Date.now() }
+  }).exec();
+  if (!user) {
+    res.status(401).json({
+      success: false,
+      message: 'Password reset token is invalid or has expired.'
+    });
+    return;
+  }
+  res.json({ success: true });
+};
+
+/**
+ * - Method: `POST`
+ * - Endpoint: `/reset-password/:token`
+ * - Authenticated: `false`
+ * - Id: `UserController.updatePassword`
+ *
+ * Description:
+ *   - Used by the new password form to update a user's password with the valid token
+ *   - Returns a Generic 401 - 'Password reset token is invalid or has expired.' if the token timed out
+ *   - Returns a PublicUser if successfully saved
+ *   - Returns an Error if network error on save attempt
+ */
+export const updatePassword: RequestHandler<
+  ResetOrUpdatePasswordRequestParams,
+  PublicUserOrErrorOrGeneric,
+  UpdatePasswordRequestBody
+> = async (req, res) => {
+  const user = await User.findOne({
+    resetPasswordToken: req.params.token,
+    resetPasswordExpires: { $gt: Date.now() }
+  }).exec();
+  if (!user) {
+    res.status(401).json({
+      success: false,
+      message: 'Password reset token is invalid or has expired.'
+    });
+    return;
+  }
+
+  user.password = req.body.password;
+  user.resetPasswordToken = undefined;
+  user.resetPasswordExpires = undefined;
+
+  await user.save();
+  req.logIn(user, (loginErr) => res.json(userResponse(req.user!)));
+  // eventually send email that the password has been reset
+};
+
+/**
+ * - Method: `PUT`
+ * - Endpoint: `/account`
+ * - Authenticated: `true`
+ * - Id: `UserController.updateSettings`
+ *
+ * Description:
+ *   - Used to update the user's username, email, or password while authenticated
+ */
+export const updateSettings: RequestHandler<
+  {},
+  PublicUserOrError,
+  UpdateSettingsRequestBody
+> = async (req, res) => {
+  try {
+    const user = await User.findById(req.user!.id);
+    if (!user) {
+      res.status(404).json({ error: 'User not found' });
+      return;
+    }
+    user.username = req.body.username;
```

Comment:
```suggestion
    user.username = req.body.username ?? user.username;
```
Or error early if req.body doesn't have username or email